src/data/resolvers/role.ts

```diff
@@ -0,0 +1,78 @@
+import { Resolvers } from '@apollo/client';
+import {
+  ClientType,
+  ColonyClientV6,
+  getMultipleEvents,
+} from '@colony/colony-js';
+import { LogDescription } from 'ethers/utils';
+
+import { Context } from '~context/index';
+import {
+  ColonySubscribedUsersQuery,
+  ColonySubscribedUsersQueryVariables,
+  ColonySubscribedUsersDocument,
+  getMinimalUser,
+} from '~data/index';
+
+const getUsersArchitectureRoles = (roleSetEvents: LogDescription[]) => {
+  const userAddresses: Record<string, boolean> = {};
+  roleSetEvents.map(({ values: { user, setTo } }) => {
+    if (setTo) {
+      userAddresses[user] = true;
+    } else {
+      userAddresses[user] = false;
+    }
+    return null;
+  });
+  return Object.keys(userAddresses).filter(
+    (userAddress) => !!userAddresses[userAddress],
+  );
+};
+
+export const roleResolvers = ({
+  colonyManager,
+  apolloClient,
+}: Required<Context>): Resolvers => ({
+  Query: {
+    async architectureRoleUsers(_, { colonyAddress }) {
+      try {
+        const subscribedUsers = await apolloClient.query<
+          ColonySubscribedUsersQuery,
+          ColonySubscribedUsersQueryVariables
+        >({
+          query: ColonySubscribedUsersDocument,
+          variables: {
+            colonyAddress,
+          },
+        });
+
+        const colonyClient = (await colonyManager.getClient(
+          ClientType.ColonyClient,
+          colonyAddress,
+        )) as ColonyClientV6;
+
+        const rolesSet = await getMultipleEvents(colonyClient, [
+          colonyClient.filters.ColonyRoleSet(null, null, null, null, null),
+        ]);
+
+        const allUsers = subscribedUsers?.data?.subscribedUsers || [];
+        const userWithArchitectureRoles = getUsersArchitectureRoles(rolesSet);
+
+        const users = userWithArchitectureRoles.map((userAddress) => {
+          const userWithProfile = allUsers.find(
+            ({ profile: { walletAddress } }) =>
+              walletAddress.toLowerCase() === userAddress.toLowerCase(),
+          );
+          if (!userWithProfile) {
+            return getMinimalUser(userAddress);
+          }
+          return userWithProfile;
+        });
+        return users;
+      } catch (error) {
+        console.error(error);
+        return [];
+      }
+    },
+  },
```

Comment:
There's a couple of issues that you need to tackle here:
- change `username` to something else (see above)
- you need to return `avatarHash`, `displayName` and `userName` for the minimal profile
- you need to check if the `subscribedUsers` data and `colonyAddress` exists before starting to use it

Here's my implementation:
```js
    async architectureRoleUsers(_, { colonyAddress }) {
      try {
        const subscribedUsers = await apolloClient.query<
          ColonySubscribedUsersQuery,
          ColonySubscribedUsersQueryVariables
        >({
          query: ColonySubscribedUsersDocument,
          variables: {
            colonyAddress,
          },
        });

        if (subscribedUsers?.data?.subscribedUsers && colonyAddress) {
          const colonyClient = (await colonyManager.getClient(
            ClientType.ColonyClient,
            colonyAddress,
          )) as ColonyClientV6;

          const rolesSet = await getMultipleEvents(colonyClient, [
            colonyClient.filters.ColonyRoleSet(null, null, null, null, null),
          ]);

          const allUsers = subscribedUsers?.data?.subscribedUsers || [];
          const userWithArchitectureRoles = getUsersArchitectureRoles(rolesSet);

          return userWithArchitectureRoles.map((userAddress) => {
            const userWithProfile = allUsers.find(
              ({ profile: { walletAddress } }) =>
                walletAddress.toLowerCase() === userAddress.toLowerCase(),
            );
            if (!userWithProfile) {
              const minimalUser = getMinimalUser(createAddress(userAddress));
              return {
                ...minimalUser,
                profile: {
                  userName: null,
                  displayName: null,
                  avatarHash: null,
                  walletAddress: minimalUser.id,
                },
              };
            }
            return {
              ...userWithProfile,
              profile: {
                ...userWithProfile.profile,
                userName: userWithProfile?.profile?.username || null,
              },
            };
          });
        }
        return [];
      } catch (error) {
        console.error(error);
        return [];
      }
    },
```

Combined with the the `username` prop rename, this works perfectly:

![Screenshot from 2021-03-22 20-48-46](https://user-images.githubusercontent.com/1193222/112042304-086df300-8b50-11eb-8e6b-1cb9bfc5f2d0.png)

Reply:
Omg, how you found out about this naming problem? Thanks!!